components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h

```diff
@@ -87,6 +87,16 @@
  */
 @property(nonatomic, strong, nullable) UIColor *topHandleColor;
 
+/**
+ The fraction of the screen to which the drawer may expand initially.
+
+ Must be between 0 (0%) and 1 (100%) inclusive. Default is 0.5 (50%).
+
+ Note: When TraitCollection is UIUserInterfaceSizeClassCompact or when using
+ VoiceOver or SwitchControl, the drawer will always present at full screen (100%).
+ */
+@property(nonatomic, assign) CGFloat maximumInitialDrawerHeightFactor;
```

Comment:
This API implies that the drawer will appear at different heights on different devices. Practically, I expect this to mean that a value chosen for an iPhone X may not be a good value for an iPhone SE.

My assumption of the expected use case for this API is to ensure that a specific amount of content is shown to the user. If this assumption is true, I then expect a client to have to conditionally evaluate the percentage based on the device's form factor or the context in which this view controller is being presented.

With the above in mind, have you considered making this API a point-based API so that one value can be chosen and behavior consistently realized across a variety of product form factors?

Reply:
Yeah this is a discussion I have had with our clients as to what would be a reasonable API to provide. 3 options came in mind:
1. Provide a percentage/factor based API based on the screen's height (which is the current implementation).
2. Provide an absolute point-based API
3. Provide a BOOL where NO = 50%, and YES = maximum height of content if smaller than full screen.

I believe each has their pros and cons:
Option 1:
Pros: 
* Consistent with how we currently present the drawer; the spec asks for a default value of 50% of the screen's height. 
* One might look at it from the perspective of how much of the content is seen, but I wonder if design (based on the 50% choice) may have looked at it from the perspective of having the drawer take up consistently a certain percentage of the screen's real estate in terms of esthetics and not blocking too much of the view behind the drawer.
* Easier to implement as we already use the notion of percentage in our implementation.
Cons: 
* Not allowing full control of showing the same amount of content for different devices.
* Can be used incorrectly if providing a percentage higher than 1 or smaller than 0 (ignoring the documentation recommendation).

Option 2:
Pros:
* Allows to consistently show the same amount of the drawer's content throughout different devices.
Cons:
* Easier to get confused on what point exactly to provide to get the desired result.
* May need some changes in code to allow the move to a point-based API and away from a percentage based API.
* A reasonable point for an iPhone X may be unreasonable for an iPhone 4.
* The default value for this API will be dynamic as it is based off a calculation that is related to Option 1, having the drawer open at 50%. This will make it very difficult to revert back to the default value as well after setting it to a point value, as those are 2 entirely different concepts.

Option 3:
Pros:
* Impossible to mess up, and very easy to set and get the expected behavior.
Cons:
* Lacks flexibility and will probably need follow-up work once new requests come in for more customization.

Based on these factors and seeing what works for our clients I chose Option 1. Although I don't strongly oppose moving to a point-based system, if you think it's the better way to go. I would have to re-sync with design and see what they think if we do move to that.

Reply:
Design's input and our clients' input will be valuable here, namely in terms of the multiple device context and what the expected behavior is. My assumption may be wrong, so I unfortunately can't say in isolation whether it's a better way to go.

Reply:
I'll gather the inputs from design and our clients and proceed from there, thanks.

Reply:
Have you considered adding a delegate API for this? Then clients can implement traitCollection behavior, per-device logic, etc.

Reply:
Also, given the 3 options presented, this may warrant a brief design doc to capture this information more permanently.

Reply:
Based on design and client feedback this has been modified to be a point-based API for customizing the drawer's height.